Exception when using "--nogui" in a never used Windows machine

When setting up new Windows server machines to use an automation system that runs IPED, an user got an exception related to the initialization of DocThumbTask.
It relies on LibreOffice, but it is not present in "C:\Users\<user>\AppData\Local\iped" as the machine was never used to run IPED and the automatic extraction fails if `--nogui` is used, which is the case.

```
2024-05-03 20:10:45     [MSG]   [processing.ui.ProgressConsole]         Inicializando DocThumbTask
2024-05-03 20:10:45     [ERROR] [app.processing.Main]                   Processing Error:
java.io.IOException: Cannot run program "null/program/soffice.bin": CreateProcess error=2, O sistema não pode encontrar o arquivo especificado
        at java.lang.ProcessBuilder.start(Unknown Source) ~[?:?]
        at java.lang.ProcessBuilder.start(Unknown Source) ~[?:?]
        at iped.engine.task.DocThumbTask.init(DocThumbTask.java:131) ~[iped-engine-4.2-snapshot.jar:?]
        at iped.engine.core.Worker.initTasks(Worker.java:128) ~[iped-engine-4.2-snapshot.jar:?]
        at iped.engine.core.Worker.<init>(Worker.java:110) ~[iped-engine-4.2-snapshot.jar:?]
        at iped.engine.core.Manager.initWorkers(Manager.java:561) ~[iped-engine-4.2-snapshot.jar:?]
        at iped.engine.core.Manager.process(Manager.java:283) ~[iped-engine-4.2-snapshot.jar:?]
        at iped.app.processing.Main.startManager(Main.java:179) [iped.jar:?]
        at iped.app.processing.Main.execute(Main.java:242) [iped.jar:?]
        at iped.app.processing.Main.main(Main.java:310) [iped.jar:?]
Caused by: java.io.IOException: CreateProcess error=2, O sistema não pode encontrar o arquivo especificado
        at java.lang.ProcessImpl.create(Native Method) ~[?:?]
        at java.lang.ProcessImpl.<init>(Unknown Source) ~[?:?]
        at java.lang.ProcessImpl.start(Unknown Source) ~[?:?]
        ... 10 more
ERROR!!!
```

A possible workaround is to manually copy libreoffice to the expected folder from another machine in which IPED was used without `--nogui`. However it is not very practical when working with several machines (and possibly different users in each machine).

--- a/iped-app/src/main/java/iped/app/bootstrap/Bootstrap.java
+++ b/iped-app/src/main/java/iped/app/bootstrap/Bootstrap.java
@@ -145,10 +145,7 @@ public class Bootstrap {
                 classpath += separator + pluginConfig.getPluginFolder().getAbsolutePath() + "/*";
             }
 
-            // user can't open analysis UI w/ --nogui, so no need to load libreoffice jars
-            if (!iped.getCmdLineArgs().isNogui()) {
-                classpath = fillClassPathWithLibreOfficeJars(iped, classpath);
-            }
+            classpath = fillClassPathWithLibreOfficeJars(iped, classpath, iped.getCmdLineArgs().isNogui());
 
             String javaBin = "java";
             if (SystemUtils.IS_OS_WINDOWS) {
@@ -278,14 +275,14 @@ public class Bootstrap {
         return props;
     }
 
-    private static String fillClassPathWithLibreOfficeJars(Main iped, String classpath)
+    private static String fillClassPathWithLibreOfficeJars(Main iped, String classpath, boolean isNogui)
             throws URISyntaxException, IOException {
         System.setProperty(IOfficeApplication.NOA_NATIVE_LIB_PATH,
                 new File(iped.getRootPath(), "lib/nativeview").getAbsolutePath());
         LibreOfficeFinder loFinder = new LibreOfficeFinder(new File(iped.getRootPath()));
-        if (loFinder.getLOPath() != null) {
+        if (loFinder.getLOPath(isNogui) != null) {
             List<File> jars = new ArrayList<>();
-            UNOLibFinder.addUNOJars(loFinder.getLOPath(), jars);
+            UNOLibFinder.addUNOJars(loFinder.getLOPath(isNogui), jars);
             for (File jar : jars) {
                 classpath += separator + jar.getCanonicalPath();
             }

--- a/iped-app/src/main/java/iped/app/ui/ViewerController.java
+++ b/iped-app/src/main/java/iped/app/ui/ViewerController.java
@@ -116,7 +116,7 @@ public class ViewerController {
                     URI jarUri = URLUtil.getURL(LibreOfficeViewer.class).toURI();
                     File moduledir = new File(jarUri).getParentFile().getParentFile();
                     LibreOfficeFinder loFinder = new LibreOfficeFinder(moduledir);
-                    final String pathLO = loFinder.getLOPath();
+                    final String pathLO = loFinder.getLOPath(false);
                     if (pathLO != null) {
                         SwingUtilities.invokeAndWait(new Runnable() {
                             @Override

--- a/iped-engine/src/main/java/iped/engine/task/DocThumbTask.java
+++ b/iped-engine/src/main/java/iped/engine/task/DocThumbTask.java
@@ -94,7 +94,7 @@ public class DocThumbTask extends ThumbTask {
                         URL url = URLUtil.getURL(this.getClass());
                         File jarDir = new File(url.toURI()).getParentFile();
                         LibreOfficeFinder loFinder = new LibreOfficeFinder(jarDir);
-                        loPath = loFinder.getLOPath();
+                        loPath = loFinder.getLOPath(true);
                         logger.info("LibreOffice Path: " + loPath);
                     }
 

--- a/iped-viewers/iped-viewers-impl/src/main/java/iped/viewers/util/LOExtractor.java
+++ b/iped-viewers/iped-viewers-impl/src/main/java/iped/viewers/util/LOExtractor.java
@@ -23,7 +23,7 @@ import iped.viewers.localization.Messages;
 public class LOExtractor extends CancelableWorker<Object, Object> {
 
     private File output, input;
-    private ProgressDialog progressMonitor;
+    private volatile ProgressDialog progressMonitor;
     private int progress = 0, numSubitens = 2876;// TODO obter número de itens automaticamente
     private boolean completed = false;
 

--- a/iped-viewers/iped-viewers-impl/src/main/java/iped/viewers/util/LOExtractor.java
+++ b/iped-viewers/iped-viewers-impl/src/main/java/iped/viewers/util/LOExtractor.java
@@ -32,7 +32,7 @@ public class LOExtractor extends CancelableWorker<Object, Object> {
         this.input = input;
     }
 
-    public boolean decompressLO() {
+    public boolean decompressLO(boolean isNogui) {
 
         try {
             if (output.exists()) {
@@ -44,14 +44,16 @@ public class LOExtractor extends CancelableWorker<Object, Object> {
             }
 
             if (input.exists()) {
-                SwingUtilities.invokeLater(new Runnable() {
-                    @Override
-                    public void run() {
-                        progressMonitor = new ProgressDialog(null, LOExtractor.this);
-                        progressMonitor.setMaximum(numSubitens);
-                        progressMonitor.setNote(Messages.getString("LOExtractor.DecompressingLO")); //$NON-NLS-1$
-                    }
-                });
+                if (!isNogui) {
+                    SwingUtilities.invokeLater(new Runnable() {
+                        @Override
+                        public void run() {
+                            progressMonitor = new ProgressDialog(null, LOExtractor.this);
+                            progressMonitor.setMaximum(numSubitens);
+                            progressMonitor.setNote(Messages.getString("LOExtractor.DecompressingLO")); //$NON-NLS-1$
+                        }
+                    });
+                }
 
                 try (ZipFile zip = new ZipFile(input)) {
                     Enumeration<ZipArchiveEntry> e = zip.getEntries();
@@ -69,15 +71,17 @@ public class LOExtractor extends CancelableWorker<Object, Object> {
                             if (++progress == numSubitens)
                                 completed = true;
 
-                            progressMonitor.setProgress(progress);
-
-                            if (progressMonitor.isCanceled())
-                                break;
+                            if (progressMonitor != null) {
+                                progressMonitor.setProgress(progress);
+    
+                                if (progressMonitor.isCanceled())
+                                    break;
+                            }
                         }
                     }
                 }
 
-                if (!progressMonitor.isCanceled()) {
+                if (progressMonitor != null && !progressMonitor.isCanceled()) {
                     progressMonitor.close();
                 }
 

--- a/iped-viewers/iped-viewers-impl/src/main/java/iped/viewers/util/LibreOfficeFinder.java
+++ b/iped-viewers/iped-viewers-impl/src/main/java/iped/viewers/util/LibreOfficeFinder.java
@@ -36,7 +36,7 @@ public class LibreOfficeFinder {
         baseDir = baseFolder;
     }
 
-    public String getLOPath() {
+    public String getLOPath(boolean isNogui) {
         if (detectedPath == null) {
             detectedPath = System.getProperty("libreOfficePath");
             if (detectedPath != null)
@@ -62,7 +62,7 @@ public class LibreOfficeFinder {
 
                 if (winPathLO.exists() || compressedLO.exists()) { // $NON-NLS-1$ //$NON-NLS-2$
                     LOExtractor extractor = new LOExtractor(compressedLO, winPathLO);
-                    if (extractor.decompressLO())
+                    if (extractor.decompressLO(isNogui))
                         detectedPath = winPathLO.getAbsolutePath();
                 }
             }

